be/src/olap/rowset/segment_v2/binary_plain_page.h

```diff
@@ -80,24 +80,21 @@ class BinaryPlainPageBuilder : public PageBuilder {
     Slice finish() override {
         _finished = true;
 
-        size_t offsets_pos = _buffer.size();
-
-        // Set up the header
-        encode_fixed32_le(&_buffer[0], _offsets.size());
-        encode_fixed32_le(&_buffer[4], offsets_pos);
+        // Set up trailer
+        size_t new_cap = _buffer.size() + _offsets.size() * sizeof(uint32_t) + MAX_TRAILER_SIZE;
+        if (new_cap > _buffer.capacity())
+            _buffer.GrowArray(new_cap);
 
         _buffer.append(&_offsets[0], _offsets.size() * sizeof(uint32_t));
```

Comment:
We should also use `put_fixed32_le` to write each offset, otherwise it doesn't write LE offset on big endian platform.

Reply:
👌

Reply:
I find that put_fixed32_le use encode_fixed32_le to write value to tmp buffer,and then append tmp buffer to dst.
And encode_fixed32_le contains the logic which judges the byte order and adjust the byte order of buffer,So I don't understand why using encode_fixed32_le will cause " it doesn't write LE offset on big endian platform"
I think the difference between encode_fixed32_le and put_fixed32_le is that caller doesn't need init a tmp buffer when calling put_fixed32_le

Reply:
Yeah, both `encode_fixed32_le` and `put_fixed32_le` handles byte order correctly. Here I'm talking about the `_offsets` vector that is copied to `_buffer` by memcpy, which doesn't handle byte order.